micmac_li3ds/api.py

```diff
@@ -402,15 +402,15 @@ def update_obj(args, metadata, obj, type_):
         obj.setdefault('description', 'Imported from "{basename}"')
     if type_ in args:
         obj.update({k: v for k, v in args[type_].items() if v})
-    for key in obj:
-        if obj[key]:
+    for key in list(obj.keys()):
+        if obj[key] and isinstance(obj[key], str):
             try:
                 obj[key] = obj[key].format(**metadata)
-            except AttributeError:
-                continue
             except KeyError as e:
-                err = 'metadata {} not available for {}/{}="{}"'
-                raise KeyError(err.format(e, type_, key, obj[key]))
+                # obj[key] contain replacements fields that have no
+                # corresponding keys in metadata, just delete the
+                # key from the object in that case
+                del obj[key]
```

Comment:
But this way you are handling the same way missing keys due to erroneous keys and missing keys which were present but with value None.

I propose to have this function handle the removal of keys with value None into a temporary metadata object (rather than in each command) and report the error in the log if the keyerror was neither in the original metadata object.